fuzz/validate.cc

```diff
@@ -0,0 +1,21 @@
+#include <fuzzer/FuzzedDataProvider.h>
+
+#include <memory>
+#include <string>
+
+#include "ada.cpp"
+#include "ada.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  FuzzedDataProvider fdp(data, size);
+  std::string source = fdp.ConsumeRandomLengthString(256);
+  std::string base_source = fdp.ConsumeRandomLengthString(256);
+
+  /**
+   * ada::validate
+   */
+  auto out_aggregator = ada::parse<ada::url_aggregator>(source);
```

Comment:
I am somewhat anxious about this code. Are you assuming that 'source' is parsed successfully?

Reply:
> I am somewhat anxious about this code. Are you assuming that 'source' is parsed successfully?

good point @lemire, I'll update it so it only validates the parsed successfully ones.

Reply:
Code updated.